### ISSUE TITLE
[GNM Temporal 9/9] ARKit paired captureをteacherにcausal dynamic priorを学習・評価する

### ISSUE THREAD

**Issue by matsuvr** [matsuvr/vrm_bevy_vtuber#68]
Parent: #50
Depends on: #57

## 目的

決定論的なdense GNM tracker + adaptive temporal priorのbaselineが完成し、Direct MediaPipeとのA/B基盤ができた後に、**開発時だけiPhone/ARKitをteacherとして使い、ARKit/Animojiに近い時間方向のmotion priorを学習する最終研究段階**を実装する。

ランタイムでiPhoneを接続する機能ではない。

最終Windowsアプリの入力は引き続き普通のWebcamだけとする。

```text
Development / dataset generation only:
  iPhone front camera RGB + ARKit Face Tracking
     -> timestamp-aligned ARKit blendshapes/head pose
     -> same RGB framesをMediaPipe/GNMへoffline replay
     -> paired sequences

Runtime target:
  ordinary Webcam
     -> MediaPipe observations
     -> persistent GNM state estimator
     -> learned causal dynamic prior / correction
     -> temporal-coherent GNM state
     -> ARKit52 decode
     -> VRM
```

## なぜ最後に行うか

いきなりblack-box temporal NNを入れると、

- dense observation改善
- identity固定
- warm start
- deterministic temporal regularization
- adaptive weight

のどれが効いているのか分離できない。

このIssueは#57のdeterministic baselineをsource of truthとして、そのbaselineを**ARKit teacherに対してどこまで改善できるか**だけを検証する。

## teacherの位置づけ

ARKit outputを「真の人間表情ground truth」とは呼ばない。

この研究でのteacherは:

> Apple ARKit/Animoji系のtime-domain behaviorへ近付けるためのtarget signal

である。

目標は人間の筋運動を物理的に完全復元することではなく、利用者がiPhone外付け顔trackingで感じる低jitter・低lag・一貫した表情trajectoryを普通のWebcamで近似すること。

## paired capture contract

収録時に最低限次を同一monotonic timelineで保持する。

```text
ArkitTeacherFrame
  timestamp
  frame_seq
  RGB frame reference / derived feature reference
  ARKit blendshape coefficients
  ARKit head transform
  optional face anchor metadata needed for alignment
```

同じRGB frameをofflineでcurrent MediaPipe pipelineへ通し、さらにGNM baselineをreplayする。

```text
PairedTemporalSample
  timestamp
  MediaPipe dense observation
  MediaPipe auxiliary blendshape observation
  deterministic GNM state
  fit residual / confidence / adaptive policy state
  baseline ARKit52 projection
  teacher ARKit52
  teacher head pose
```

異なるframe/timestampをnearestだけで雑に結合しない。capture/decode latencyとtimestamp domainを明示する。

## RGB / privacy policy

顔動画はbiometric-sensitiveな研究データとして扱う。

- raw RGB/videoをgitへcommitしない。
- repository fixtureへ個人顔画像を入れない。
- dataset pathをgitignore対象またはrepository外にする。
- 可能ならMediaPipe/GNM feature extraction後にraw RGBを削除できるworkflowを用意する。
- derived numeric traceだけでtraining/replayできる形を優先する。
- dataset metadataに氏名等の不要な個人情報を保存しない。

third-party人物データは、明確な利用権なしに収録/配布しない。

## runtimeにiPhone依存を持ち込まない

禁止:

- app起動時にiPhone/ARKit connectionを要求
- Live Link Face等をproduction dependencyにする
- ARKit runtime streamがないとGNM modeを使えない構造
- teacher capture protocolを通常UIの必須機能にする

teacher capture helperは開発/研究ツールとして隔離する。

## dynamic priorの学習対象

最初から `RGB -> ARKit52` のend-to-end置換器を作らない。

GNMをcanonical face stateとして維持する。

第一候補は、直近数frameのGNM state / motion / confidence / residualから、次frame fitに使う**prior mean / predicted velocity / bounded correction**を推定するcausal model。

概念:

```text
history:
  GNM state_{t-k..t-1}
  velocities
  observation quality
  motion score
  residuals
        |
        v
CausalDynamicPrior
        |
        +--> predicted expression/pose prior
        +--> uncertainty / weight suggestion
        |
        v
GNM fitting at t
        |
        v
ARKit52 baseline/projector
```

ARKit teacher lossは最終ARKit52 trajectoryとの差を使えるが、runtime state authorityはGNM fitting側に残す。

## model候補 / complexity policy

いきなりTransformerを前提にしない。

候補を軽い順に比較する。

1. linear state-space / autoregressive model
2. MPPCA / mixture linear dynamics
3. small causal TCN
4. small GRU
5. それでも不足する場合だけより複雑なsequence model

採用条件:

- held-out sequenceでdeterministic adaptive baselineより改善
- real-time budget内
- bounded memory
- causal（未来frameを参照しない）
- export/inference contractが明確

modelサイズの大きさ自体を品質とみなさない。

## training/runtime dependency boundary

production applicationは引き続きRust-onlyを原則とする。

- Python runtime禁止
- Python sidecar禁止
- normal `cargo build/test`がPythonを要求しない

ただし研究用offline trainingについて、PyTorch/JAX等が明確に最小実装となる場合は、**通常build/runtimeから完全に分離されたoptional research tool**としてPython使用を許容してよい。

その場合:

- production appはPythonを呼ばない。
- training environment/versionを別文書へ固定する。
- exported model artifactのschema/version/hashを明示する。
- production inference方法とlicenseを別途監査する。
- repositoryの通常テストはprecomputed synthetic/reference model fixtureだけで完結する。

Rustだけで十分な学習方式なら、その方を優先してよい。

## teacher loss

単純なper-frame MAEだけではARKit的smoothnessを学べない。

最低限、候補loss/metricsを分離して扱う。

```text
L_value       // coefficient value agreement
L_velocity    // first derivative agreement
L_accel       // second derivative / trajectory shape
L_pose        // head pose temporal agreement
L_onset       // onset/rise timing
L_regularize  // correction magnitude / bounded prior
```

全項を必須とはしない。ablationで必要性を示し、採用式を記録する。

teacherを追うためにdense landmark reprojectionを無視して顔stateを幻覚することは禁止する。GNM data termとのbalanceを保持する。

## causality / latency

offline evaluationでも未来frameを入力に使わない。

- centered moving average
- bidirectional RNN
- future framesを使うsmoother

はruntime比較対象として不可。

look-back historyは許容するが、その分のbuffer/latencyはreportする。

predictionを使ってcamera latencyを補償する場合も、overshoot/false motionを測定する。

## held-out evaluation

training sequenceと評価sequenceを分離する。

可能なら次を分ける。

- neutral/static
- slow expression
- fast smile/release
- blink / eye motion
- mouth/jaw speech-like motion
- head yaw/pitch/roll
- combined pose + expression
- partial occlusion / tracking degradation

同一連続takeを細切れにしてtrain/testへ混ぜ、実質duplicateにしない。

人物数が1人だけの初期研究なら、その制約を明記し「general person model」と主張しない。identity固定/normalizationによる一般化は別途検証する。

## evaluation targets

#57のDirect/Deterministic GNM baselineに加えて、ARKit teacherを第三の参照としてreportする。

最低限:

- per-channel value MAE/correlation
- velocity error
- acceleration/jerk error
- stationary jitter relative to ARKit
- onset delay relative to ARKit
- 10/50/90% rise time difference
- peak attenuation/overshoot
- blink-like pulse timing
- head pose temporal error
- dropout/reacquire behavior
- runtime inference latency
- model memory/CPU/GPU cost

「ARKitとのMAEだけ小さいが100ms遅れる」は成功扱いしない。

## hallucination / safety constraints

learned priorは観測が弱い時に顔を勝手に大きく動かしてはならない。

- correction magnitudeをboundedにする。
- low confidence時はuncertaintyを出せる構造を優先する。
- no-face/lost時のlifecycleは#64/#66 authorityを維持する。
- training range外のstateでNaN/Infを出さない。
- learned model failure時はdeterministic adaptive baselineへfallbackできる。

## A/B/Ablation

最低限次を同じheld-out traceで比較する。

1. Direct MediaPipe
2. GNM persistent + no temporal
3. GNM fixed temporal
4. GNM adaptive temporal
5. GNM adaptive + learned dynamic prior
6. ARKit teacher

これにより「learned priorが本当に必要か」を判断できる。

改善が小さい/latencyが増える場合、learned runtimeを採用しない結論も正当な完了結果とする。研究Issueなので、モデル採用ありきにしない。

## artifact/versioning

learned modelを採用する場合:

- model format/schema version
- training config hash
- source code revision
- teacher dataset metadata hash（raw biometric dataそのものではない）
- exported model SHA-256
- inference compatibility version

を記録する。

opaque model fileだけを説明なしにcommitしない。再配布/licenseが不明なteacher-derived artifactをreleaseへ入れない。

## 必須テスト

1. paired sampleのtimestamp/sequence mismatchを拒否する。
2. teacher ARKit valuesのcanonical 52 orderをPR #51 contractへ正規化する。
3. offline replayで同じRGB/derived traceから同じMediaPipe/GNM sequenceを再現できる範囲を固定する。
4. future frameをruntime model inputへ使わない。
5. model output/correctionがfinite/bounded。
6. model unavailable/corrupt/version mismatchでdeterministic baselineへfallback。
7. low-confidence/lost stateでlarge hallucinated expressionをpublishしない。
8. held-out evaluationがtrain traceと分離される。
9. Direct / deterministic GNM / learned / ARKit teacherのreportを生成できる。
10. velocity/acceleration/onset metricsを含む。
11. learned prior追加でruntime latencyを測定できる。
12. raw face video/imageをrepository fixtureへcommitしない。
13. normal workspace build/testがPython/iPhoneを要求しない。
14. production runtimeでiPhone connectionを要求しない。

## acceptance

- development-only ARKit paired capture/replay contractがある。
- same RGB-derived observationsとARKit teacherをtime-alignedに比較できる。
- GNM canonical stateを維持したcausal dynamic priorを最低1方式評価する。
- deterministic adaptive baselineとのablationがある。
- ARKit teacherに対しvalueだけでなくvelocity/acceleration/onset/jitterを比較する。
- learned modelはcausal/bounded/fallback可能。
- ordinary Webcamだけのproduction runtimeを維持する。
- Pythonを使う場合もoffline research toolingに隔離し、production/runtime/build/testへ持ち込まない。
- raw biometric footageをgitへ入れない。
- learned modelがbaselineを十分改善しない場合、採用しない結論を許容する。
- GitHub Actionsを使わない。

## 非対象

- iPhoneをproduction mocap deviceとして使う機能
- Face ID/TrueDepth raw depthの製品依存
- cloud dataset upload
- third-party biometric dataset収集
- end-to-end RGB->ARKit black-boxへのGNM置換
- non-causal offline smoothingをproduction品質として扱うこと

## ローカル検証

通常repository gate:

```text
cargo fmt --all -- --check
cargo check --workspace
cargo test --workspace --no-fail-fast
cargo clippy --workspace --all-targets -- -D warnings
cargo deny check
git diff --check
```

teacher capture/trainingは対応環境で別途実行し、exact command/config/hash/resultを研究reportへ記録する。実行できないcaptureを成功と書かない。
